samples_project/Assets/SampleViewer/Scripts/SampleSwitcher.cs

```diff
@@ -179,10 +175,95 @@ private IEnumerator PipelineChanged()
 
         yield return null;
 
-        SetPipeline(PipelineTypeDropdown.options[PipelineTypeDropdown.value].text);
+        // Set the Pipeline based on what Pipeline button is clicked
+        SetPipeline(PipelineModeText);
+
+        SceneButtonOnClick();
+    }
+
+    public void PipelineButtonOnClick()
+    {
+        StartCoroutine(PipelineChanged());
+    }
+
+    // Delay pop-up notification
+    IEnumerator SlideNotification()
+    {
+        //Wait for 2 secs.
+        yield return new WaitForSeconds(2);
+
+        //Play notification menu animation.
+        anim.Play("NotificationAnim");
+    }
+
+    // Keep scene buttons pressed after selection
+    public void OnSceneButtonClicked(Button clickedBtn)
+    {
+        int btnIndex = System.Array.IndexOf(sceneButtons, clickedBtn);
+
+        if (btnIndex == -1)
+            return;
+
+        foreach (Button btn in sceneButtons)
+        {
+            btn.interactable = true;
+        }
+
+        clickedBtn.interactable = false;
+    }
+
+    // Keep pipeline buttons pressed after selection
+    public void OnPipelineButtonClicked(Button clickedBtn)
+    {
+        int btnIndex = System.Array.IndexOf(pipelineButtons, clickedBtn);
+
+        if (btnIndex == -1)
+            return;
 
-        SceneChanged();
+        foreach (Button btn in pipelineButtons)
+        {
+            btn.interactable = true;
+        }
+
+        clickedBtn.interactable = false;
     }
+
+    // Set HDRP button color
+    public void SetHDRPColor()
```

Comment:
These four functions (`SetBlahColor` and `UnloadBlahColor`) could be refactored to a single function. It looks like outside editor UI might be calling these, so you could keep the functions, but have them wrap a unified function.

```
public void SetHDRPColor()
{
    SetPipelineColor(0, pipelineButtons[0].colors.selectedColor, false);
}

public void SetPipelineColor(int index, Color color, bool active)
{
        var colors = pipelineButtons[index].colors;
        colors.normalColor = color;
        pipelineButtons[index].colors = colors;
        pipelineButtons[index].interactable = active;
}
```

Reply:
Good call! Thanks! I change the `SetSthColor` to the suggested unified method.

@NaumanIsGISMan for `UnloadSthColor`, I used two separate buttons to call each method. Can I still use a unified method? Or I should keep `UnloadSthColor` as the way it is?

![image](https://github.com/Esri/arcgis-maps-sdk-unity-samples/assets/106696949/b1c805d3-e521-4bd3-801e-c89dd1563a62)

Reply:
Good call! Thanks!